app/lib/widgets/settings/share_settings.dart

```diff
@@ -0,0 +1,200 @@
+import 'dart:developer';
+import 'package:flutter/cupertino.dart';
+import 'package:mooovy/bike/bike.dart';
+import 'package:flutter/material.dart';
+import 'package:mooovy/local_storage.dart';
+import 'package:email_validator/email_validator.dart';
+import 'package:mooovy/widgets/settings/section.dart';
+
+class ShareBike extends StatefulWidget {
+  const ShareBike({required this.bike, super.key});
+  final Bike bike;
+
+  @override
+  State<ShareBike> createState() => _ShareWith();
+}
+
+class _ShareWith extends State<ShareBike> {
+  final GlobalKey<FormState> _formKey = GlobalKey<FormState>();
+  final api = obtainApiClient();
+
+  double _duration = 14;
+  String _email = '';
+  String? error;
+  String? success;
+
+  submitForm(NavigatorState navigator) async {
+    if (_formKey.currentState?.validate() != true) return;
+    _formKey.currentState?.save();
+    if (_email.isEmpty) return;
+
+    setState(() {
+      error = null;
+      success = null;
+    });
+
+    try {
+      int dur = _duration.toInt() * 86400;
+
+      var res = await api?.shareCurrentBike(widget.bike.id, _email, dur);
+
+      setState(() {
+        success = 'Shared with ${res["email"]}';
+      });
+    } catch (e) {
+      setState(() {
+        error = e.toString();
+      });
+    }
```

Comment:
If we in either way do a state update why not call state update after the try catch
```suggestion
      success = 'Shared with ${res["email"]}';
    } catch (e) {
      error = e.toString();
    }
    setState(() {});
```